### PR TITLE
DbaDatabaseSnapshot commands - pipeline support

### PR DESCRIPTION
**Pull request by ctrlbold**

### CODE REVIEW

File: functions/Get-DbaDatabaseSnapshot.ps1 [hunk 1]
```diff
@@ -16,8 +16,8 @@ Credential object used to connect to the SQL Server as a different user
 .PARAMETER Databases
 Return information for only specific base dbs
 
-.PARAMETER Exclude
-Return information for all but these specific base dbs
+.PARAMETER Snapshots
```
Comment (PowerDBAKlaas) [inline functions/Get-DbaDatabaseSnapshot.ps1]: even if multiple items can be submitted, the parameter name should be singular.

File: functions/Get-DbaDatabaseSnapshot.ps1 [hunk 2]
```diff
@@ -55,18 +54,20 @@ Returns informations for database snapshots excluding ones that have HR as base
 		[string[]]$SqlServer,
```
Comment (PowerDBAKlaas) [inline functions/Get-DbaDatabaseSnapshot.ps1]: '$SqlInstance' will be our parameter name
'sqlserver' can be an alias

File: functions/Get-DbaDatabaseSnapshot.ps1 [hunk 3]
```diff
@@ -55,18 +54,20 @@ Returns informations for database snapshots excluding ones that have HR as base
...
+	
 	BEGIN
 	{
 		# Convert from RuntimeDefinedParameter object to regular array
 		$databases = $psboundparameters.Databases
-		$exclude = $psboundparameters.Exclude
+		$snapshots = $psboundparameters.Snapshots
 	}
 
 	PROCESS
```
Comment (PowerDBAKlaas) [inline functions/Get-DbaDatabaseSnapshot.ps1]: foreach ($Instance in $SqlInstance)

will be our variable of choice from now on

File: functions/Get-DbaDatabaseSnapshot.ps1 [hunk 4]
```diff
@@ -77,42 +78,46 @@ Returns informations for database snapshots excluding ones that have HR as base
...
-					DatabaseCreated = $db.createDate
 					SnapshotOf = $db.DatabaseSnapshotBaseName
+					SizeMB = [Math]::Round($db.Size,2)
+					DatabaseCreated = $db.createDate
+					IsReadCommittedSnapshotOn = $db.IsReadCommittedSnapshotOn
+					SnapshotIsolationState = $db.SnapshotIsolationState
+					SnapshotDb = $db
 				}
+				
+				Select-DefaultField -InputObject $object -Property Server, Database, SnapshotOf, SizeMB, DatabaseCreated, IsReadCommittedSnapshotOn, SnapshotIsolationState
```
Comment (PowerDBAKlaas) [inline functions/Get-DbaDatabaseSnapshot.ps1]: shrewd :-)

File: functions/Remove-DbaDatabaseSnapshot.ps1 [hunk 5]
```diff
@@ -56,10 +59,15 @@ Remove-DbaDatabaseSnapshot -SqlServer sqlserver2014a -Databases HR, Accounting
 Removes all database snapshots having HR and Accounting as base dbs
 
 .EXAMPLE
-Remove-DbaDatabaseSnapshot -SqlServer sqlserver2014a -Exclude HR
+Remove-DbaDatabaseSnapshot -SqlServer sqlserver2014a -Snapshots HR_snapshot, Accouting_snapshot
```
Comment (PowerDBAKlaas) [inline functions/Remove-DbaDatabaseSnapshot.ps1]: typo: "accouting" (2x)

File: functions/Get-DbaDatabaseSnapshot.ps1 [hunk 6]
```diff
@@ -42,10 +42,9 @@ Get-DbaDatabaseSnapshot -SqlServer sqlserver2014a -Databases HR, Accounting
 Returns informations for database snapshots having HR and Accounting as base dbs
 
 .EXAMPLE
-Get-DbaDatabaseSnapshot -SqlServer sqlserver2014a -Exclude HR
-
-Returns informations for database snapshots excluding ones that have HR as base dbs
+Get-DbaDatabaseSnapshot -SqlServer sqlserver2014a -Snapshots HR_snapshot, Accouting_snapshot
```
Comment (PowerDBAKlaas) [inline functions/Get-DbaDatabaseSnapshot.ps1]: typo: "accouting" (2x)

File: functions/Remove-DbaDatabaseSnapshot.ps1 [hunk 7]
```diff
@@ -68,79 +76,113 @@ Removes all database snapshots excluding ones that have HR as base dbs
 		[Alias("ServerInstance", "SqlInstance")]
 		[string[]]$SqlServer,
```
Comment (PowerDBAKlaas) [inline functions/Remove-DbaDatabaseSnapshot.ps1]: parameter name '$SqlInstance'
alias 'SqlServer'

File: functions/Remove-DbaDatabaseSnapshot.ps1 [hunk 8]
```diff
@@ -68,79 +76,113 @@ Removes all database snapshots excluding ones that have HR as base dbs
...
+					Database = $PipelineSnapshot.Database
+					SnapshotOf = $PipelineSnapshot.SnapshotOf
+					Status = $status
+				}
+			}
+			return
+		}
+		
+		# if piped value either doesn't exist or is not the proper type
 		foreach ($servername in $SqlServer)
```
Comment (PowerDBAKlaas) [inline functions/Remove-DbaDatabaseSnapshot.ps1]: foreach ($Instance in $SqlInstance)

File: functions/Get-DbaDatabaseSnapshot.ps1 (hunk 1 above)
Comment (ctrlbold) [inline functions/Get-DbaDatabaseSnapshot.ps1]: Yes, usually. But I tried singular and didn't like it. It didn't convey what I wanted to SQL people, which is that singular params can do arrays.

File: functions/Get-DbaDatabaseSnapshot.ps1 (hunk 4 above)
Comment (ctrlbold) [inline functions/Get-DbaDatabaseSnapshot.ps1]: It'll be even better when I implement -Exclude just gotta find the time ;)

**Comment by ctrlbold** [comment]
Closing to redo with no additional files.

**Pull request closed without merging**